controllers/hash.php

```diff
@@ -1,7 +1,34 @@
 <?php
+function getHash() {
```

Comment:
@event15 @efiku  Sure, go ahead and improve my PR. The branch is available. Or start a new branch. But keep in mind, should work without a git command available. And bonus points for being able to use a Fallback.

To be honest. The reason for this code is that git for windows is not a real requirement and many people here definitely are not using git as a command line tool. This was just playing nice...